dScripts/QbSpawner.cpp

```diff
@@ -0,0 +1,144 @@
+#include "QbSpawner.h"
+#include "BaseCombatAIComponent.h"
+#include "MovementAIComponent.h"
+#include "dLogger.h"
+
+void QbSpawner::OnStartup(Entity* self) {
+	auto mobNum = self->GetVar<int>(u"mobNum");
+	auto spawnDist = self->GetVar<float>(u"spawnDist");
+	auto mobTemplate = self->GetVar<LWOOBJID>(u"mobTemplate");
+	auto spawnTime = self->GetVar<float>(u"spawnTime");
+
+	if (!mobNum) self->SetVar<int>(u"mobNum", m_DefaultMobNum);
+	if (!spawnDist) self->SetVar<float>(u"spawnDist", m_DefaultSpawnDist);
+	if (!mobTemplate) self->SetVar<LWOOBJID>(u"mobTemplate", m_DefaultMobTemplate);
+	if (!spawnTime) self->SetVar<float>(u"spawnTime", m_DefaultSpawnTime);
+
+	// go ahead and setup the mob table here
+	std::vector<LWOOBJID> mobTable;
+	mobTable.assign(self->GetVar<int>(u"mobNum"), LWOOBJID_EMPTY);
+
+	self->SetVar<std::vector<LWOOBJID>>(u"mobTable", mobTable);
+}
+
+void QbSpawner::OnFireEventServerSide(Entity* self, Entity* sender, std::string args, int32_t param1, int32_t param2, int32_t param3) {
+	Game::logger->Log("QbSpawner", "FireEVernt");
+	auto gateObjID = sender->GetObjectID();
+	if (!gateObjID) return;
+	if (args == "spawnMobs") {
+		self->SetVar(u"gateObj", gateObjID);
+		auto spawnTime = self->GetVar<float>(u"spawnTime");
+		self->AddTimer("SpawnMobEnemies", spawnTime);
+	}
+}
+
+void QbSpawner::OnTimerDone(Entity* self, std::string timerName) {
+	if (timerName == "SpawnMobEnemies") {
+		auto mobTable = self->GetVar<std::vector<LWOOBJID>>(u"mobTable");
+
+		auto spawnDist = self->GetVar<float>(u"spawnDist");
+		auto mobTemplate = self->GetVar<LWOOBJID>(u"mobTemplate");
+
+		auto gateObjID = self->GetVar<LWOOBJID>(u"gateObj");
+		if (!gateObjID) return;
+
+		auto* gate = EntityManager::Instance()->GetEntity(gateObjID);
+		if (!gate) return;
+
+		auto oPos = gate->GetPosition();
+		auto oDir = gate->GetRotation().GetForwardVector();
+		auto newPos = NiPoint3::ZERO;
+		newPos.x = oPos.x + (oDir.x * spawnDist);
+		newPos.y = oPos.y;
+		newPos.z = oPos.z + (oDir.z * spawnDist);
```

Comment:
This should be contatenated to `NiPoint3 newPos(oPos.x + (oDir.x * spawnDist), oPos.y, oPos.z + (oDir.z * spawnDist));`